### PR TITLE
Ensure virt-export route has redirect policy

### PR DESCRIPTION
**Pull request by awels**
### What this PR does
Before this PR:
The virt-export route did not have an EdgeTerminationPolicy set.

After this PR:
The virt-export route has an EdgeTerminationPolicy of redirect to ensure that insecure requests are automatically redirected to the secure route.

Fixes https://issues.redhat.com/browse/CNV-43248

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Virt export route has an edge termination of redirect
```

### CODE REVIEW

File: pkg/virt-operator/resource/generate/components/routes_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package components
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-operator/resource/generate/components/routes_test.go:1]: Should it be `components_test`?

**Review by fossedihelm — COMMENTED** [review]
Thank you @awels!
General question: Should we redirect or block insecure connections?
Regarding the code: only two little things regarding the pkg.

I know that this is outside of the scope of this PR but I think that only one of `GetAllRoutes` and `NewExportProxyRoute` should be exported (maybe the first one).
Currently, `NewExportProxyRoute` is exported to allow the usage in the `kubevirt_test` file, which does not look so good to me :)

**Comment by awels** [comment]
Either block or redirect are options. I am just following what other similar routes do, and they do redirect so I did the same. Let me add the header and change the package name.

**Comment by fossedihelm** [comment]
> Either block or redirect are options. I am just following what other similar routes do, and they do redirect so I did the same. Let me add the header and change the package name.

Fair enough! Thank you!
/lgtm

**Comment by alicefr** [comment]
Looks good, thanks!
/approve

**Pull request merged**